AK/Iterator.h

```diff
@@ -16,6 +16,7 @@ class SimpleIterator {
     friend Container;
 
     constexpr bool is_end() const { return m_index == SimpleIterator::end(m_container).m_index; }
+    constexpr bool past_end() const { return m_index >= SimpleIterator::end(m_container).m_index; }
```

Comment:
I use it in a few places to handle the case where I parse to the end, and return to the middle of a for loop. While I agree that this is not super pretty, It's just as wrong to dereference an iterator that is at the end and one that is past the end. I could easily rework the PR to not use this, but it would make the parser uglier, by adding:
```c++
if (tokens.at_end())
    break;
```
to the bottoms of a few for loops or replacing the for loops with while loops and only incrementing in certain paths.

Reply:
The subparsers can hit the end of the iterator and then after returning to the middle of a loop, the iterator will be incremented again. Anyway, I have decided to fix this behavior instead, and to remove `past_end()`

Reply:
`past_end()` has been removed!